### PR TITLE
Add xml upload page with errors and spinner page

### PR DESCRIPTION
**Pull request by DGajewska**
WATER-1930

Add get routes for xml upload and spinner pages, post for submitting xml to water service. File is scanned and sent to water service once successful

### CODE REVIEW

File: src/views/nunjucks/returns/processing-upload.njk [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{% extends "./nunjucks/layout.njk" %}
+
+{% block content %}
+  <h1 class="govuk-heading-xl">{{ pageTitle }}</h1>
+  <p class="govuk-body-l">Please wait. This may take up to a minute.</p>
+  <div style="display: flex; justify-content: center;">
```
Comment (james-siteclick) [inline src/views/nunjucks/returns/processing-upload.njk]: We should probably avoid inline styles and create a `spinner` component in src/assets/sass/v2/blocks

File: test/modules/returns/lib/upload-helpers.js [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+  experiment('getFile', () => {
+    test('it should return a filepath to an xml file with a uuid filename', async () => {
+      const file = uploadHelpers.getFile();
+      expect(file).to.be.a.string().and.to.contain(['/temp/', '.xml']);
+    });
+  });
+  experiment('applyFormError', () => {
+    test('Form is returned with no error message if no error', async () => {
+      const error = undefined;
+      expect(await uploadHelpers.applyFormError(form, error, errorMessages)).to.equal(form);
```
Comment (james-siteclick) [inline test/modules/returns/lib/upload-helpers.js]: Could we split the async calls onto their own lines?

File: test/modules/returns/lib/upload-helpers.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+const { experiment, test, beforeEach, afterEach } = exports.lab = require('lab').script();
+const sinon = require('sinon');
+const fs = require('fs');
+const EventEmitter = require('events');
+
+const uploadHelpers = require('../../../../src/modules/returns/lib/upload-helpers.js');
+const fileCheck = require('../../../../src/lib/file-check');
+
+const csrfToken = '4a0b2424-6c02-45a5-9935-70a4c41538d2';
+const errorMessages = {
```
Comment (james-siteclick) [inline test/modules/returns/lib/upload-helpers.js]: I think these are defined in the controller and could be imported here

File: src/modules/returns/controllers/upload.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+const getSpinnerPage = async (request, h) => {
+  const view = {
+    ...request.view,
+    pageTitle: 'Uploading returns data'
+  };
+
+  const eventId = request.params.event_id;
+
+  // Get data from event database
+  const { data, error } = await water.events.findOne(eventId);
```
[the remark this answers is not in the document]
Comment (DGajewska) [reply src/modules/returns/controllers/upload.js]: I made a few changes to this, having it log an error if no data is returned, as well as redirecting the user back to the upload page and stating that there was an upload error

**Pull request merged**